api/v1alpha1/machinedeletionremediation_types.go

```diff
@@ -20,6 +20,13 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// ProcessingConditionType is the condition type used to signal NHC the remediation status
```

Comment:
This comes directly from [SNR](https://github.com/medik8s/self-node-remediation/blob/2365c8bc37bc0a0ab2868ab88a89827f85c2e728/api/v1alpha1/selfnoderemediation_types.go#L26-L27)
However, it does a bit more than that, it signals whether the remediation has ever started (!UNKNOWN), is processing (STARTED), or has ended. So a more generic "status" seems good to me.

My initial intention was to add the "known values" there as well (they also appeared during SNR implementation, but then removed in the final version), but since they are defined elsewhere I decided not to.
I can add them anyway if you think it's better.

Reply:
The "known values" are added automatically to the CRD file when you introduce the `Conditions []metav1.Condition`  in this file. I don't see any harm in adding it another time in the code (even thought it might propagate to other files), but it is minor thing and up to you.